test/unit/Chainweb/Test/Pact5/RemotePactTest.hs

```diff
@@ -145,112 +162,134 @@ mkFixture v baseRdb = do
         , _serviceClientEnv = serviceClientEnv
         }
 
+class HasFixture where
+    remotePactTestFixture :: IO Fixture
+
+withFixture' :: IO Fixture -> ((CutFixture.HasFixture, HasFixture) => a) -> a
+withFixture' fixture tests =
+    withDict @HasFixture fixture $
+    CutFixture.withFixture' (_cutFixture <$> remotePactTestFixture) tests
+
+withSharedFixture :: ResourceT IO Fixture -> ((CutFixture.HasFixture, HasFixture) => TestTree) -> TestTree
+withSharedFixture mk tests =
+    withResourceT mk $ \fixture ->
+        withFixture' fixture tests
+
+withFixture :: Fixture -> ((CutFixture.HasFixture, HasFixture) => a) -> a
+withFixture fixture tests = withFixture' (return fixture) tests
+
+-- generating this cert and making an HTTP manager take quite a while relative
+-- to the rest of the tests, so they're shared globally.
+-- there's no apparent reason to ever switch them out, either.
+cert :: X509CertPem
+key :: X509KeyPem
+(_, cert, key) = unsafePerformIO $ generateLocalhostCertificate @RsaCert 1
+defaultTLSSettings :: HTTP.TLSSettings
+defaultTLSSettings = (HTTP.TLSSettingsSimple True False False TLS.defaultSupported)
+httpManager :: Manager
+httpManager = unsafePerformIO $ HTTP.newTlsManagerWith (HTTP.mkManagerSettings defaultTLSSettings Nothing)
```

Comment:
Lawful good usage of `unsafePerformIO`

Reply:
:pray: I would have avoided this if tasty didn't wrap resources in IO, which would be very cumbersome.